### ISSUE TITLE
Draggable, after switching between tabs does not call onDragEnd

### ISSUE THREAD

**Issue by karezza** [flutter/flutter#55985]
## Steps to Reproduce

1. Here is the code, copy into a project & load up the Screen
2. Drag the first ListTile which says "NEW" in the same way you would a reorderablelistview to the right side of the screen, which should cause the tab to switch.
3. Let go of the drag and you will not see console out put 'on drag end'.
4. If you retry, and just drag around the initial tab and release it will say 'on drag end'.

```
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';
import 'dart:async';

class KanbanScreen extends StatelessWidget {
  final int projectId;

  KanbanScreen({Key key, @required this.projectId}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: WKanban(projectId),
    );
  }
}

class Toggle {
  bool state;
  final _controller = StreamController<bool>();

  Toggle(bool state) {
    state ? state = false : state = true;

    print("add to sink: ${!state}");
    _controller.sink.add(state);
  }

  trigger() {
    print("add to sink: ${state}");
    _controller.sink.add(state);
  }

  Stream<bool> get stream => _controller.stream;
}

class KanbanState extends State<WKanban> with SingleTickerProviderStateMixin/*, AutomaticKeepAliveClientMixin*/ {
  TabController tabController;

  Toggle toggle;
  int _selected;

  @override
  void initState() {
    super.initState();

    tabController = TabController(vsync: this, length: 6);

    toggle = Toggle(false);
    _selected = 0;
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
        appBar: AppBar(
            title: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: <Widget>[
            Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: <Widget>[
                  Text("ProjectName"),
                  Text("KANBAN", style: TextStyle(fontSize: 16.0)),
                ]),
          ],
        )),
        body:
            Row(
                //crossAxisAlignment: CrossAxisAlignment.start,
                children: <Widget>[
              Text("["),
              Expanded(
                child: Scaffold(
                  appBar: AppBar(
                    leading: Container(),
                    flexibleSpace: SafeArea(
                      child: TabBar(controller: tabController, tabs: [
                        Tab(text: "New"),
                        Tab(text: "Ready"),
                        Tab(text: "In Progress"),
                        Tab(text: "Test"),
                        Tab(text: "Done"),
                        Tab(text: "Archived"),
                      ]),
                    ),
                  ),
                  body:
                      TabBarView(controller: tabController, children: <Widget>[
                    ListView(
                        //onReorder: (one, two) {},
                        children: <Widget>[
                          Draggable<Color>(
                            data: Color(0x000000ff),
                            child: ListTile(
                              key: Key("a"),
                              leading:
                                  Text("NEW", style: TextStyle(fontSize: 20.0)),
                              trailing:
                                  Text("10", style: TextStyle(fontSize: 20.0)),
                            ),
                            feedback: Text("dragging"), // when dragging
                            childWhenDragging:
                                Text("from"), // what's left behind
                            onDragStarted: () {
                              print(
                                  "drag has started, make DragTargets visible");
                              print("showDragTargets = true");
                              toggle.trigger();
                            },
                            onDragEnd: (details) {
                              print("on drag end, make DragTargets invisible");
                              toggle.trigger();
                            },
                          ),
                          ListTile(key: Key("b"), title: Text("111")),
                          ListTile(key: Key("c"), title: Text("222")),
                          ListTile(key: Key("d"), title: Text("333")),
                        ]),
                    ReorderableListView(
                        onReorder: (one, two) {},
                        children: <Widget>[
                          ListTile(
                            key: Key("z"),
                            leading:
                                Text("READY", style: TextStyle(fontSize: 20.0)),
                            trailing:
                                Text("20", style: TextStyle(fontSize: 20.0)),
                          ),
                          ListTile(key: Key("y"), title: Text("aaa")),
                          ListTile(key: Key("x"), title: Text("bbb")),
                          ListTile(key: Key("w"), title: Text("ccc")),
                        ]),
                    Text("In Progress"),
                    Text("Ready for Test"),
                    Text("Done"),
                    Text("Archived"),
                  ]),
                ),
                //),
              ),
              StreamBuilder<bool>(
                  stream: toggle.stream,
                  builder: (context, snapshot) {
                    if ((snapshot.connectionState == ConnectionState.active) &&
                        snapshot.hasData)
                      return Visibility(
                        visible: snapshot.data,
                        child: DragTarget<Color>(
                            //key: Key("asdf"),
                            onWillAccept: (value) => value != Colors.black26,
                            onAccept: (value) {
                              print("drag target onAccept");
                            },
                            onLeave: (value) {
                              print("drag target onLeave");
                            },
                            builder: (context, candidates, rejects) {
                              if (candidates.length > 0) {
                                // move to next tab to right, avoid out-of-bounds
                                _selected++;
                                if (_selected == 6) {
                                  _selected = 5;
                                } else {
                                  Future changeTab = Future(() {
                                    tabController.animateTo(_selected);
                                  });
                                }
                              }
                              return (candidates.length > 0)
                                  ? Container(
                                      color: Colors.green,
                                      child: Center(
                                          child: Column(children: <Widget>[
                                        Text("done"),
                                      ])))
                                  : Container(
                                      color: Colors.green,
                                      child: Center(
                                          child: Column(children: <Widget>[
                                        Text(">>>>"),
                                      ])));
                            }),
                      );

                    return Container();
                  }),
            ]));
  }
}

class WKanban extends StatefulWidget {
  final int projectId;

  WKanban(this.projectId);

  @override
  KanbanState createState() => KanbanState();
}
```

**Comment by karezza** [comment]
```
C:\Users\travis\AndroidStudioProjects\taiga_mobile>flutter doctor -v
[√] Flutter (Channel stable, v1.12.13+hotfix.9, on Microsoft Windows [Version 10.0.17763.1098], locale en-US)
    • Flutter version 1.12.13+hotfix.9 at C:\src\flutter
    • Framework revision f139b11009 (4 weeks ago), 2020-03-30 13:57:30 -0700
    • Engine revision af51afceb8
    • Dart version 2.7.2


[!] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at C:\Users\travis\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)
    X Android license status unknown.
      Try re-installing or updating your Android SDK Manager.
      See https://developer.android.com/studio/#downloads or visit https://flutter.dev/setup/#android-setup for detailed instructions.

[√] Android Studio (version 3.6)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)

[√] Connected device (1 available)
    • SM G960U1 • 4939584f54563398 • android-arm64 • Android 10 (API 29)

! Doctor found issues in 1 category.
```

**Comment by iapicca** [comment]
Hi @karezza 
I'm not quite sure that the intended behaviour 
of a `Draggable` inside a `TabBarView`
is to cause the tab  to switch

This platform is not meant for assistance on personal code.
Please see https://flutter.dev/community for resources and asking questions like this,
you may also get some help if you post it on Stack Overflow.

Closing, as this isn't an issue with Flutter itself,
if you disagree please write in the comments and I will reopen it
possibly providing a **minimal** reproducible code sample
Thank you

**Comment by karezza** [comment]
Hello,

1. Your summary does not match the bug I was intending to report.  I was not trying to get a Draggable to cause a tab switch to occur.  Instead, I have created the situation where you are dragging a item between tabs.  I do use a drag target to trigger the tab switching, but I could have also used a timer.  - As the draggable continues to drag around after the tab is switched, as intended, seems that upon releasing the draggable would call the onDragEnd callback as expected.

2. I'm not looking for help on personal code, I have already done my due diligence with searching the web and chat channels.  It could be that the solution is something like AutomaticKeepAliveClientMixin but at this point I think I have genuinely identified a bug, hence the bug report.

3. I felt the code I submitted with the original was pretty minimal.  Since there are tabs and listviews I have a few tabs and a few list items, if you want me to modify to have two tabs and only one item in each list view I can make that change ... otherwise there is just enough to create the needed items and switch tabs while dragging just the first item on the first tab to the tab on the right.
